clang/test/dpct/sharedmem-dim-noUSM.cu

```diff
@@ -83,6 +83,9 @@ __global__ void kernel3() {
 }
 
 void foo3() {
+  // CHECK: /*
+  // CHECK-NEXT: DPCT1099:{{[0-9]+}}: The size of the accessor is a literal value to avoid potential out of scope issue of the used macro/const value.
```

Comment:
Size of the accessor in this case is not a macro, there is no "out-of-scope" problem with using original expression in this case? Why do we need to replace it with value and provide warning?

Reply:
The `sizeof` can not be keep in some cases and it hard to idetify the scenarios. For example:
```c++
__global__ void foo29();
void foo30(){
  foo29<<<1,1,0>>>();
}
class A{};
__global__ void foo29() {
  __shared__ double red_acc[sizeof(A)];
}
```
There will be "out-of-scope" problem if we don't replace the expression. (because class A is invisible in function `foo30`)
But if the expression only contain unary/binary/ternary expression consist of integer literal, we can keep the expression not changed, and I prefer to fix it in a new PR.